lib/cpu.js

```diff
@@ -0,0 +1,91 @@
+var os = require('os')
+var fs = require('fs')
+var exec = require('child_process').exec
+
+/**
+ * Gathers Clock, PageSize and system uptime through /proc/uptime
+ * This method is mocked in procfile tests
+ */
+function updateCpu (cpu, next) {
+  if (cpu !== null) {
+    getRealUptime(function (err, uptime) {
+      if (err) return next(err)
+      cpu.uptime = uptime
+      next(null, cpu)
+    })
+    return
+  }
+
+  getClockAndPageSize(function (err, data) {
+    if (err) return next(err)
+
+    getRealUptime(function (err, uptime) {
+      if (err) return next(err)
+
+      cpu = {
+        clockTick: data.clockTick,
+        pageSize: data.pageSize,
+        uptime: uptime
+      }
+
+      next(null, cpu)
+    })
+  })
+}
+
+module.exports = updateCpu
+
+function getconf (keyword, options, next) {
+  if (typeof options === 'function') {
+    next = options
+    options = { default: '' }
+  }
+
+  exec('getconf ' + keyword, function (error, stdout, stderr) {
+    if (error !== null) {
+      console.error('Error while getting ' + keyword, error)
+      return next(null, options.default)
+    }
+
+    stdout = parseInt(stdout)
+
+    if (!isNaN(stdout)) {
+      return next(null, stdout)
+    }
+
+    return next(null, options.default)
+  })
+}
+
+function getClockAndPageSize (next) {
+  getconf('CLK_TCK', {default: 100}, function (err, clockTick) {
```

Comment:
The two `getconf ` shouldn't be called parallelly?

Reply:
Lazy to add a flow library only for this. In an ideal world yes but as this is called only once it's fine IMO.

Reply:
in fact going to add this haha